### ISSUE TITLE
links not converted correctly (macOS)

### ISSUE THREAD

**Issue by fibretothepremises** [zoni/obsidian-export#17]
## Obsidian flavor markdown
```md
# 0.1.md
[[0.2|Create new site]]
```
0.1.md and 0.2.md are in the same directory.

## Converted markdown
```md
# 0.1.md
[Create new site](0.2.md)
```
This will not render properly as this implies there is a file in path `0.1/0.2` when they are in the same parent directory.

The correct path should render as below:
```md
# 0.1.md
[Create new site](../0.2.md)
```

Please reply to this or ping me if this will be fixed.
Thanks for your work.

**Comment by zoni** [comment]
Hi @fibretothepremises, thanks for raising this.

Unfortunately, the behavior you describe is intended and entirely correct. Given two (markdown) files in the same directory, the correct path to link from one to the other is `[title](file.md)`, not `[title](../file.md)` as that would imply you're linking to a file that is one directory above the current file.

However, I suspect you are using some kind of static site generator to publish the results, and this is turning a file `/dir/foo.md` into a URL like `/dir/foo/` and `/dir/bar.md` into `/dir/bar/`, is that right?

In that case foo and bar indeed end up in different directories from each other. This could be worked around by introducing an extra `../` in the links in the source markdown files, but that's addressing the problem in the wrong place. It's **up to the static site generator to account for this** and change these links accordingly.

If you happen to be using Hugo, you're in luck. Check out these resources for more information and a fix:

- https://notes.nick.groenen.me/notes/relative-linking-in-hugo/
- https://github.com/zoni/obsidian-export/issues/8#issuecomment-774521792

**Comment by fibretothepremises** [comment]
Thanks for the quick reply.

It's funny you are directing me to render hooks as I found your program as an alternative to render hooks. Now, if I go the render hooks route I'm not sure what the purpose of obsidian-export would be. Is it designed for other site builders? (I use Hugo). 

Am I missing something here? I thought this program was built so that we don't have to use render hooks?

Thanks again. I have tried copying and pasting your code, however, to no avail. I'm beginning to think Hugo is way too much effort for a personal blog.

Cheers
